File: output/CodeLlama/codenet/C++/Java/s973217665.java
import java.util.Scanner;
 
public class s973217665 {
    public static void main(String[] args) {
        Scanner in = new Scanner(System.in);
        int n = in.nextInt();
        while (n-- > 0) {
            int p = in.nextInt();
            int q = in.nextInt();
            int l = (int) Math.sqrt(p * p + q * q);
            boolean isprime = true;
            for (int i = 0; i < l; i++) {
                for (int j = 0; j < l; j++) {
                    int x = i * i + j * j;
                    if (x <= 1 || x >= p * p + q * q) continue;
                    if ((i * p + j * q) % x == 0 && (i * q - j * p) % x == 0) isprime = false;
                }
            }
            System.out.println(isprime ? "P" : "C");
        }
    }
}

File: output/ChatGPT/codenet/C++/Python/s973217665.py
import math

def is_prime(p, q):
    l = int(math.sqrt(p**2 + q**2))
    isprime = True
    for i in range(l):
        for j in range(l):
            x = i**2 + j**2
            if x <= 1 or x >= p**2 + q**2:
                continue
            if (i*p + j*q) % x == 0 and (i*q - j*p) % x == 0:
                isprime = False
    return isprime

n = int(input())
for _ in range(n):
    p, q = map(int, input().split())
    result = "P" if is_prime(p, q) else "C"
    print(result)